concreate/generator.py

```diff
@@ -40,13 +44,17 @@ def prepare_modules(self, descriptor=None):
         if not descriptor:
             descriptor = self.descriptor
 
+        modules = descriptor.get('modules', {}).get('install', {})
+
         # If descriptor doesn't requires any module we can start merging descriptors
-        # and fetching artifacts. There ibs nothing left to do except for this
-        if 'modules' not in descriptor:
+        # and fetching artifacts. There is nothing left to do except for this
+        if not modules:
```

Comment:
this can be else branch for for cycle after your re factoring which is more readable

Reply:
NACK